File: backend/backend/src/main/java/com/jj/backend/entity/Project.java
package com.jj.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "projects")
@Inheritance(strategy = InheritanceType.JOINED)
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "company_name")
    private String companyName;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @ManyToMany(fetch = FetchType.EAGER, cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinTable(name = "projects_technologies", joinColumns = @JoinColumn(name = "project_id", referencedColumnName = "id"), inverseJoinColumns = @JoinColumn(name = "technology_id", referencedColumnName = "id"))
    private List<Technology> technologies = new ArrayList<>();

    @OneToMany(mappedBy = "project")
    Set<UserProject> users;
}

File: frontend/src/models/Project.ts
import type { TechnologyResponse } from "./technology";
import type { StandardUserProjectResponse } from "./user";

export type ProjectRequest = {
  userId: number;
  name: string;
  companyName: string;
  description: string;
  projectRole: string;
  startDate: string;
  endDate: string;
  technologies: number[];
};

export type Project = {
  projectId: number;
  name: string;
  companyName: string;
  description: string;
  startDate: string;
  endDate: string;
  createAt: string;
  updatedAt: string;
  users: StandardUserProjectResponse[];
  technologies: TechnologyResponse[];
};
